packages/sync-repo-settings/src/sync-repo-settings.ts

```diff
@@ -83,15 +62,39 @@ function handler(app: Application) {
     const name = context.payload.repository.name;
     const repo = `${owner}/${name}`;
 
-    // find the repo record in repos.json
-    const repos = await handler.getRepos();
-    const yoshiRepo = repos.repos.find(x => x.repo === repo);
-    console.log(yoshiRepo?.repo);
-    if (!yoshiRepo) {
+    // Fetch the list of languages used in this repository
+    const langRes = await context.github.repos.listLanguages({
+      owner,
+      repo: name,
+    });
+
+    // Given an object like this:
+    // {
+    //   python: 24
+    //   javascript: 22
+    // }
+    // Sort the keys based on instance count.
+    const languages = Object.entries(langRes.data).sort((a, b) => {
+      return a[1] > b[1] ? -1 : a[1] < b[1] ? 1 : 0;
+    });
+
+    // If GitHub says this doesn't have a language ...
+    if (languages.length === 0) {
       return;
     }
-    if (languageConfig[yoshiRepo.language]) {
-      const ignored = languageConfig[yoshiRepo.language].ignoredRepos?.find(
+
+    // Use the language with the highest line count
+    let language = languages[0][0].toLowerCase();
+
+    // Add a little hackery for node
+    if (language === 'javascript' || language === 'typescript') {
+      language = 'nodejs';
+    }
+    console.log(`Determined ${repo} is ${language}`);
+
```

Comment:
should we add some use cases *just in case* Github returns a language we either don't support/don't recognize for some reason? or, at least logging out a message saying which language we returned for the repo?

Reply:
On second thought, I guess it should just error out in case the language is not supported. We'll at least know what language it is based on the logging!